### PR TITLE
Fix Error code in get sns attributes

### PR DESCRIPTION
**Pull request by usmangani1**
Fix Error code in get sns attributes

### CODE REVIEW

File: tests/integration/test_sns.py [hunk 1]
```diff
@@ -848,3 +848,35 @@ def test_create_topic_after_delete_with_new_tags(self):
...
+
+        subscription_attributes = self.sns_client.get_subscription_attributes(
+            SubscriptionArn=subscription['SubscriptionArn'])
+
+        self.assertEqual(subscription_attributes.get('Attributes').get('SubscriptionArn'),
+                         subscription['SubscriptionArn'])
+
+        self.sns_client.unsubscribe(SubscriptionArn=subscription['SubscriptionArn'])
+
+        try:
```
Comment (whummer) [inline tests/integration/test_sns.py:872]: nit: We can follow the `assertRaises` pattern provided by the Python `unittest` library:
```
with self.assertRaises(Exception) as ctx:
    self.sns_client.get_subscription_attributes(
        SubscriptionArn=subscription['SubscriptionArn'])
self.assertEqual(ctx.exception.response['Error']['Code'], 'NotFound')
self.assertEqual(ctx.exception.response['ResponseMetadata']['HTTPStatusCode'], 404)
```

**Review by whummer — COMMENTED** [review]
LGTM, thanks @usmangani1 👍  - only a minor suggestion for refactoring the test logic. Thanks

File: tests/integration/test_sns.py (hunk 1 above)
Comment (usmangani1) [reply tests/integration/test_sns.py]: Resolved

File: tests/integration/test_sns.py [hunk 2]
```diff
@@ -848,3 +848,35 @@ def test_create_topic_after_delete_with_new_tags(self):
...
+
+        self.assertEqual(subscription_attributes.get('Attributes').get('SubscriptionArn'),
+                         subscription['SubscriptionArn'])
+
+        self.sns_client.unsubscribe(SubscriptionArn=subscription['SubscriptionArn'])
+
+        with self.assertRaises(ClientError) as ctx:
+            self.sns_client.get_subscription_attributes(
+                SubscriptionArn=subscription['SubscriptionArn'])
+            self.fail('This call should not be successful as the subscription does not exist')
```
Comment (whummer) [inline tests/integration/test_sns.py:875]: We can remove this line (sorry, should have made this more clear..). Thanks

Comment (usmangani1) [reply tests/integration/test_sns.py]: Done

**Pull request merged**